saleor/order/models.py

```diff
@@ -234,15 +234,13 @@ def ship(self, tracking_number=''):
     def cancel(self):
         cancel_delivery_group(self)
 
-    @property
-    def shipping_required(self):
-        return self.shipping_method_name is not None
-
     def get_total_quantity(self):
         return sum([line.get_quantity() for line in self])
 
     def is_shipping_required(self):
-        return self.shipping_required
+        return any([
+            line.product.product_type.is_shipping_required
+            if line.product else True for line in self.lines.all()])
```

Comment:
I'm not sure about assuming `True` and relying on nullable `product` here. I think the safest solution would be to add `OrderLine.is_shipping_required` field and set its value based on `CartLine.is_shipping_required()` when creating groups.